homeassistant/components/trafikverket_weatherstation/sensor.py

```diff
@@ -197,29 +210,36 @@ class TrafikverketWeatherStation(SensorEntity):
 
     def __init__(
         self,
-        weather_api,
-        name,
-        sensor_station,
+        weather_api: TrafikverketWeather,
+        name: str,
+        sensor_station: str,
```

Comment:
It seems both `name` and `sensor_station` are set to the same:
```py
    sensor_name = entry.data[CONF_STATION]
    sensor_station = entry.data[CONF_STATION]
```

Is this intentional?

Reply:
That is intentional as per review before to not ask the user about the name and instead just use the Station as also the name for the entry.

Reply:
Right, and since it's no longer used, drop the name from the constructor and update:
```py
self._attr_name = f"{sensor_station} {description.name}"
```